stackrox-container-image-scanner/src/main/java/com/stackrox/jenkins/plugins/StackroxBuilder.java

```diff
@@ -252,9 +256,19 @@ public FormValidation doTestConnection(@QueryParameter("portalAddress") final St
                 if (checkRoxAuthStatus(portalAddress, apiToken, tlsVerify, caCertPEM)) {
                     return FormValidation.ok("Success");
                 }
-                return FormValidation.error(Messages.StackroxBuilder_TestConnectionError());
-            } catch (Exception e) {
-                return FormValidation.error(e, Messages.StackroxBuilder_TestConnectionError());
+                return FormValidation.error("Invalid credentials, user not authenticated");
+            } catch (Exception ex) {
+                Throwable e = Throwables.getRootCause(ex);
+                if (e instanceof ServiceException) {
+                    return FormValidation.error(e, "Invalid response from central");
+                } else if (e instanceof UnknownHostException) {
+                    return FormValidation.error(e, "Unknown host " + portalAddress);
+                } else if (e instanceof SSLException) {
+                    return FormValidation.error(e, "Could not validate SSL");
```

Comment:
```suggestion
                    return FormValidation.error(e, "Could not validate TLS");
```
It's 21st century : )